Technical-Steering-Committee/best-practices/build-and-test.md

```diff
@@ -0,0 +1,17 @@
+# Build and Test
+
+## Build
+If a Zowe component or a repository requires source code compilation, this repository should contain automated build system able to compile the software from source, automatically handle dependencies and produce installation packages. The build tools in use should be only FLOSS, free to acquire and use for any purposes. It is recommended to select tools already popular within Zowe, such as Gradle, npm, GitHub Actions, etc. For specific needs, it is possible to select alternatives, given they provide unique and necessary advantages.
+
+## Test
+Every Zowe component or repository should contain an automated test suit that is executed on every change or code build. Ideally the automated test suit should consist of the following categories of tests - unit tests, integration tests and system tests. The tests should cover both functional, i.e. validating a feature functionality, as well as non-functional, i.e. performance, reliability, resilience, etc. 
+
+## Code Analysis
+It is recommended, whenever reasonable, each repository to make use of static code analysis and vulnerability scans per tools availability and requirement. For repositories that contain mainly Java and/or Javascript code, for example, the recommendation is to enable Sonarcloud scans for each change. All reported issues should be addressed either by resolving them or by indicating an exception justification.
+
+## Quality Gate
+Every change made in Zowe component or a repository should only be introduced together with a suit of corresponding tests. Every change made in a Zowe component or a repository will require the execution of any and all relevant regression testing in depending or dependable components. A change cannot be introduced if any of the new or existing tests fail.
```

Comment:
We actually require 80% of coverage on all code that is part of Zowe. So for this part we can be more strict here.